spikeinterface/toolkit/postprocessing/principal_component.py

```diff
@@ -407,6 +418,10 @@ def _fit_by_channel_global(self):
             wfs = we.get_waveforms(unit_id)
             if wfs.size == 0:
                 continue
+
+            if p['sparsity'] is not None:
+                channel_ids = we.recording.ids_to_indices(p['sparsity'][unit_id])
```

Comment:
So this should be named `channel_inds`.

I think this trick don't work with the loop 
`for chan_ind, chan_id in enumerate(channel_ids):`

Reply:
Lets dicscuss with a call.